application/src/main/java/org/opentripplanner/transit/api/model/FilterValueCollection.java

```diff
@@ -0,0 +1,63 @@
+package org.opentripplanner.transit.api.model;
+
+import com.beust.jcommander.internal.Nullable;
+import java.util.Collection;
+import java.util.NoSuchElementException;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.service.TransitService;
+
+/**
+ * {@link FilterValueCollection} is meant to be used when filtering results from {@link TransitService}.
+ * </p>
+ * This abstraction over the Collection type lets us keep filter specific functionality separate
+ * from interpretation of various states of a collection. For instance in which case the filter values
+ * should match all entities they are meant to filter.
+ * </p>
+ * Note: Currently the only implementation of this class has functionality to include everything if the
+ * collection is empty or null. Future implementations could not include everything if the collection
+ * is empty or null, or alternatively contain more complex logic to determine if the collection should
+ * include everything.
+ * @param <E> - The type of the filter values. Typically, String or {@link FeedScopedId}.
+ */
+public class FilterValueCollection<E> {
+
+  @Nullable
+  private final Collection<E> filterValues;
+
+  FilterValueCollection(@Nullable Collection<E> filterValues) {
+    this.filterValues = filterValues;
+  }
+
+  /**
+   * Returns a {@link FilterValueCollection} that matches everything if there are no filter values.
+   * </p>
+   * @param filterValues - The {@link Collection} of filter values.
+   * @return FilterValueCollection
+   * @param <E> - The type of the filter values. Typically, String or {@link FeedScopedId}.
+   */
+  public static <E> FilterValueCollection<E> ofEmptyIsEverything(
+    @Nullable Collection<E> filterValues
+  ) {
+    return new FilterValueCollection<>(filterValues);
```

Comment:
How about setting the filterValues to null if they are empty in this constructor?

```
if (filterValues != null && filterValues.isEmpty()) {
  return new FilterValueCollection<>(null);
}
```

Then the internal state would be more explicit and `filterValues == null` would be equivalent to `includeEverything`.

Reply:
I would be ok with that, even if it breaks the basic principle of not changing the input when mapping into the POJO for the domain request.

The long term goal I want is that the API do as little as possible, and that we can add x-cutting konserns on the domain API - not API Endpoint. This is a best practice and we will reuse the feature across APIs.

An example of where this is to find all queries witch pass in empty collections. If we have trace logging(a x-cutting konsern) on the domain API, then we can find these - if we convert in the API we can not.

Reply:
If you'd rather keep it as it is, I'm fine with that as well. It just felt strange to me that the logic for "empty is everything" was put in the `includeEverything()` getter instead of in the factory method.

Reply:
I'd rather keep this as the way it is.